bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/ContentFragmentUtils.java

```diff
@@ -245,19 +252,36 @@ public static String getGridResourceType(Resource fragmentResource) {
      * @param resourceIterator Iterator of resources for which to get the component exporters.
      * @param modelFactory Model factory service.
      * @param slingHttpServletRequest Current request.
+     * @param callerResource The page or template resource that references the experience fragment or content fragment.
      * @return Ordered map of resource names to {@link ComponentExporter} models.
      */
     @NotNull
     public static LinkedHashMap<String, ComponentExporter> getComponentExporters(
         @NotNull final Iterator<Resource> resourceIterator,
         @NotNull final ModelFactory modelFactory,
-        @NotNull final SlingHttpServletRequest slingHttpServletRequest) {
+        @NotNull final SlingHttpServletRequest slingHttpServletRequest,
+        @NotNull final Resource callerResource) {
         final LinkedHashMap<String, ComponentExporter> componentExporterMap = new LinkedHashMap<>();
 
+        SlingHttpServletRequest wrappedSlingHttpServletRequest = new SlingHttpServletRequestWrapper(slingHttpServletRequest) {
+
```

Comment:
Instead of ovewriting the getAttribute method, wouldn't it be easier to set the attribute in the request? E.g.:
```
        String resourceCallerPath = (String) slingHttpServletRequest.getAttribute(ATTR_RESOURCE_CALLER_PATH);
        SlingHttpServletRequest wrappedSlingHttpServletRequest = new SlingHttpServletRequestWrapper(slingHttpServletRequest);
        if (StringUtils.isEmpty(resourceCallerPath)) {
            wrappedSlingHttpServletRequest.setAttribute(ATTR_RESOURCE_CALLER_PATH, callerResource.getPath());
        }
```

Reply:
The advantage of using a request wrapper and overriding `getAttribute` is that the attribute will get automatically removed when the original request is unwrapped, thus no need to clean up (which is something we should always do).